/bin/sh: ./build/test.sh: not found

Running `make test` produces:

```
/bin/sh: ./build/test.sh: not found
make: *** [Makefile:237: test] Error 127
- exit 2
```

This is related to #894, as can be seen in its [build job](https://github.com/kubernetes/git-sync/actions/runs/9604287488/job/26489441148).

--- a/build/build.sh
+++ b/build/build.sh
@@ -16,7 +16,6 @@
 
 set -o errexit
 set -o nounset
-set -o pipefail
 
 if [ -z "${ARCH:-}" ]; then
     echo "ARCH must be set"
@@ -35,7 +34,7 @@ export CGO_ENABLED=0
 export GOARCH="${ARCH}"
 export GOOS="${OS}"
 
-if [[ "${BUILD_DEBUG:-}" == 1 ]]; then
+if [ "${BUILD_DEBUG:-}" -eq 1 ]; then
     # Debugging - disable optimizations and inlining
     gogcflags="all=-N -l"
     goasmflags=""

--- a/build/build.sh
+++ b/build/build.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Copyright 2016 The Kubernetes Authors.
 #

--- a/build/test.sh
+++ b/build/test.sh
@@ -16,7 +16,6 @@
 
 set -o errexit
 set -o nounset
-set -o pipefail
 
 export CGO_ENABLED=0
 export GOFLAGS="-mod=vendor"
@@ -25,7 +24,7 @@ echo "Running tests:"
 go test -installsuffix "static" "$@"
 echo
 
-echo -n "Checking gofmt: "
+echo "Checking gofmt: "
 ERRS="$(go fmt "$@")"
 if [ -n "${ERRS}" ]; then
     echo "FAIL - the following files need to be gofmt'ed:"
@@ -38,7 +37,7 @@ fi
 echo "PASS"
 echo
 
-echo -n "Checking go vet: "
+echo "Checking go vet: "
 ERRS="$(go vet "$@" 2>&1 || true)"
 if [ -n "${ERRS}" ]; then
     echo "FAIL"

--- a/build/test.sh
+++ b/build/test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Copyright 2016 The Kubernetes Authors.
 #

